_posts/2024-07-03-oidc-cient-request.adoc

```diff
@@ -0,0 +1,447 @@
+```
+oc project <projectname>
+```
+
+Once you have your environment set up with the required tools, we can move on to the next step to build and deploy our application on OpenShift.
+
+== Start Keycloak
+We will be using Keycloak as our OpenID identity provider.
+
+To start a Keycloak server in your project on OpenShift, use the following command:
+
+```
+oc process -f https://raw.githubusercontent.com/keycloak/keycloak-quickstarts/latest/openshift/keycloak.yaml \
+    -p KEYCLOAK_ADMIN=admin \                 // <1>
+    -p KEYCLOAK_ADMIN_PASSWORD=admin \        // <2>
+    -p NAMESPACE=<PROJECT_NAME> \                // <3>
+| oc create -f -
+```
+
+<1> Replace admin with the user name you would like to use when accessing the Keycloak Administration Console.
+<2> Replace admin with the password you would like to use when accessing the Keycloak Administration Console.
+<3> Replace <PROJECT_NAME> with your project name.
+After running the above command, you should see the following output:
+
+```
+service/keycloak created
+route.route.openshift.io/keycloak created
+Warning: apps.openshift.io/v1 DeploymentConfig is deprecated in v4.14+, unavailable in v4.10000+
+deploymentconfig.apps.openshift.io/keycloak created.
+```
+It will take a few minutes for OpenShift to provision the Keycloak pod and its related resources.
+
+You can use the OpenShift CLI or the OpenShift web console, depending on your preference, to check if your Keycloak server has been provisioned.
+
+=== OpenShift CLI
+To make sure your Keycloak server has been provisioned using the OpenShift CLI, run:
+```
+oc get pods
+```
+
+After a little while, check for a message similar to the following message that indicates the pod is ready:
+```
+NAME                READY     STATUS      RESTARTS   AGE
+keycloak-1-deploy   0/1       Completed   0          1h
+keycloak-1-l9kdx    1/1       Running     0          1h
+```
+
+Once the Keycloak server has been provisioned, use the following command to find the URL for your Keycloak instance’s Admin Console:
+```
+KEYCLOAK_URL=https://$(oc get route keycloak --template='{{ .spec.host }}') &&
+echo "" &&
+echo "Keycloak Admin Console:   $KEYCLOAK_URL/admin" &&
+echo ""
+```
+
+=== OpenShift Web Console
+To make sure your Keycloak server has been provisioned using the OpenShift web console, navigate to the *Topology* view in the *Developer* perspective. You can click on your *keycloak* app to check its status. Once it is running, you can click on *Open URL* and you will be taken to Keycloak’s *Administration Console* login page.
+
+== Setting up Keycloak OpenID Provider
+
+. Log into the Keycloak Admin Console using the admin username and password you specified earlier. 
+. Next, create a realm called *myrealm* and register a client called *myclient* as follows: 
+*General settings*:
```

Comment:
The current Keycloak version is 25.0.1.  The screen presentation appears differently from what you describe. 
There is no "General settings" displayed anywhere.   Here are directions for the current Keycloak UI.

-- Create a realm for our test
* Select the menu in the upper left corner
* Click on the drop down box with text "Keycloak"
* Click "Create realm" button
* In the "Realm name" field type, "myrealm"
* Click the "Create" button

-- Create client, myclient
* Select the menu in the upper left corner
* Select the "Client" menu item.
* On the Client page, select the "Create client" button
* Set the "Client type" field to *OpenID Connect*
* Set the "Client ID" to *myclient*
* Select the "Next" button
* On the "Capability config" screen
** Toggle the "Client authentication" to ON
** For "Authentication flow"  select checkboxes, *Standard flow*, and *Direct access grants*
** Click the "Next" button.
* On the "Login settings" screen
** For the *Valid Redirect URIs*, enter a * for now. We will come back to edit it later.
** Click the "Save" button
* On the "Client details" page for "myclient", click the "Keys" tab.
** click on the *Import* button located at the bottom of the screen.
*** Under *Archive format*, choose *PKCS12* from the dropdown. 
*** Under *Key alias*, enter the alias of the RSA key you just created. 
*** Under *Store password* enter the password of the keystore you created
    and click on the *Browse* button to import the keystore file. Note that 
    when we created the keystore, we specified a key
    password and a keystore password. Here, we are only using the 
    *keystore* password. While we set them both to be the same, they 
    do not have to be.
*** Once you select the keystore file created earlier, click *Import* 
    and you should see a message at the top of the screen indicating 
    that the certificate has been uploaded successfully and you will 
    see it in the text field in the middle of the screen.